### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by amanramoliya**
##What

- [x] Skeleton for frontend
- [x] Github actions for test and lint

# Checklist

- [ ] My code follows the style guidelines of this project and is properly formatted
- [ ] I have performed a self-review of my own changes
- [ ] All existing tests pass, and I have added new ones if required

### CODE REVIEW

**Review by AshPrajapati — APPROVED** [review]
Requirements satisfied.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./frontend
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "14"
+      - run: npm i --save-dev @testing-library/react
```
Comment (utsav00) [inline .github/workflows/test.yml:15]: 1. Run tests should verify if the project is built is successfully or not. Check if it is included in the test step or does a build step is required to be added before test explicitly.
2. You don't have to install testing package, just `npm i` should do as long as it is in you `package.json`

File: .github/workflows/eslint.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./frontend
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "14"
+      - run: npm i eslint
```
Comment (utsav00) [inline .github/workflows/eslint.yml:15]: Same thing here - you can just `npm i` instead of installing the specific package

**Comment by utsav00** [comment]
Add a `nvmrc` file

File: .github/workflows/test.yml (hunk 1 above)
Comment (amanramoliya) [reply .github/workflows/test.yml]: Modified

File: .github/workflows/eslint.yml (hunk 2 above)
Comment (amanramoliya) [reply .github/workflows/eslint.yml]: Done

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "14"
+      - name: Install depedencies
+        run: npm i
+
+      - name: Run Build
+        id: build
+        run: npm build
```
Comment (utsav00) [inline .github/workflows/test.yml:20]: There is no need for `npm build` to run twice in both the actions. If the build succeeds once then it'll succeed second time as well. You can create another action for build and have these two actions depend on that one to remove redundant check

**Comment by utsav00** [comment]
I don't see the lint and test actions running after the build step was successful here.
Look into this for dependent jobs - https://docs.github.com/en/actions/using-workflows/about-workflows#creating-dependent-jobs and https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow#defining-prerequisite-jobs
Hope this helps

**Pull request merged**